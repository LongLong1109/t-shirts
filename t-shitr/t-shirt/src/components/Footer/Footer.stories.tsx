import type { Meta, StoryObj } from '@storybook/react'

import Footer from '.'

const meta = {
  title: 'Components/Footer',
  component: Footer,
  parameters: {
    layout: 'centered',
  },
} satisfies Meta<typeof Footer>

export default meta
type Story = StoryObj<typeof Footer>

export const FooterPrimary: Story = {
  args: {},
}
