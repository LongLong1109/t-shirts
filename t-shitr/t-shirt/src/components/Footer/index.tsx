import Link from 'next/link'

// constants
import { navData } from '@/constants/navData'

// icons
import { LogoIcon } from '@/icons'

// components
import { Paragraph, Button } from '@/components/common'

const Footer = () => (
  <footer className='flex bg-dark-75 py-20 px-32 w-full justify-between h-[369px]'>
    <div className='w-1/3'>
      <div className='flex items-center gap-2 pb-6'>
        <LogoIcon />
        <Paragraph size='xs' text='mangcoding Store' weight='bold' color='white-100' />
      </div>
      <Paragraph
        size='xs'
        color='white-100'
        text='Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam mollis, justo nec porttitor auctor, erat sapien faucibus lectus, vel tempor dolor augue et lectus. '
      />
      <div className='pt-10'>
        <Button type='submit' text='Login Now' />
      </div>
    </div>

    <div className='flex gap-20'>
      {navData.map((section, index) => (
        <div key={index}>
          <Paragraph text={section.title} size='sm' color='white-100' weight='bold' />
          {section.links.length > 0 && (
            <nav className='pt-6'>
              <ul className='flex gap-3 flex-col'>
                {section.links.map((link, linkIndex) => (
                  <li key={linkIndex}>
                    <Link className='text-white-100 font-thin text-xs' href={link.url}>
                      {link.name}
                    </Link>
                  </li>
                ))}
              </ul>
            </nav>
          )}
          {section.contact && (
            <address className='pt-6 not-italic'>
              <Paragraph text={section.contact} size='xs' color='white-100' />
            </address>
          )}
          {section.social && (
            <div>
              <nav className='list-none flex gap-2 pt-4'>
                {section.social.map((socialLink, socialIndex) => (
                  <li key={socialIndex}>
                    <Link href={socialLink.url}>{socialLink.icon}</Link>
                  </li>
                ))}
              </nav>
            </div>
          )}
        </div>
      ))}
    </div>
  </footer>
)

export default Footer
