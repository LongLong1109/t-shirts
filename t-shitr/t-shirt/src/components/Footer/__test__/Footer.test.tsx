import { render, screen } from '@testing-library/react';

import Footer from '..';

describe('Footer Component', () => {
  it('should renders correctly and matches snapshot', () => {
    const { asFragment } = render(<Footer />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should renders the mangcoding Store logo and text', () => {
    render(<Footer />);
    expect(screen.getByText('mangcoding Store')).toBeInTheDocument();
  });

  it('should renders the footer links correctly', () => {
    render(<Footer />);

    const categories = screen.getAllByText('Categories')[0];
    const shop = screen.getAllByText('Shop')[0];

    expect(screen.getByText('Product')).toBeInTheDocument();
    expect(categories).toBeInTheDocument();
    expect(shop).toBeInTheDocument();
    expect(screen.getByText('Login')).toBeInTheDocument();

    expect(screen.getByText('T-shirt')).toBeInTheDocument();
    expect(screen.getByText('Jacket')).toBeInTheDocument();
    expect(screen.getByText('Shirt')).toBeInTheDocument();
    expect(screen.getByText('Jeans')).toBeInTheDocument();

    expect(screen.getByText('Men')).toBeInTheDocument();
    expect(screen.getByText('Children')).toBeInTheDocument();
    expect(screen.getByText('Women')).toBeInTheDocument();

    expect(screen.getByText('mangcoding@gmail.com')).toBeInTheDocument();
  });
});
