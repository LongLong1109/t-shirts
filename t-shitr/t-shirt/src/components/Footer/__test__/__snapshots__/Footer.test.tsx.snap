// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component should renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <footer
    class="flex bg-dark-75 py-20 px-32 w-full justify-between"
  >
    <div
      class="w-1/3"
    >
      <div
        class="flex items-center gap-2 pb-6"
      >
        <svg
          fill="none"
          height="34"
          viewBox="0 0 36 34"
          width="36"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M35.0463 0H0.214311C0.0960349 0 0 0.0964564 0 0.215448V33.7193C0 33.8383 0.0960349 33.9347 0.214311 33.9347H35.0463C35.1646 33.9347 35.26 33.8383 35.26 33.7193V0.215448C35.26 0.0964564 35.1646 0 35.0463 0Z"
            fill="#0EC5D7"
          />
          <path
            d="M9.31885 7.46802V26.4666"
            stroke="white"
            stroke-width="3.08178"
          />
          <path
            d="M15.0122 7.46802V16.9314"
            stroke="white"
            stroke-width="3.07593"
          />
          <path
            d="M19.3486 7.46875V10.6831H24.781L28.7218 18.826H32.27L26.6346 7.46875H19.3486Z"
            fill="white"
          />
          <path
            d="M20.9629 14.1665V26.8438"
            stroke="white"
            stroke-width="3.56014"
          />
        </svg>
        <p
          class="text-xs font-bold text-white-100"
        >
          mangcoding Store
        </p>
      </div>
      <p
        class="text-xs font-undefined text-white-90"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam mollis, justo nec porttitor auctor, erat sapien faucibus lectus, vel tempor dolor augue et lectus. 
      </p>
      <div
        class="pt-10"
      >
        <button
          class="bg-white-100 px-12 py-2 h-12 text-purpleLinear-100 font-bold"
          type="submit"
        >
          Login Now
        </button>
      </div>
    </div>
    <div
      class="flex gap-20"
    >
      <div>
        <p
          class="text-sm font-bold text-white-100"
        >
          Home
        </p>
        <nav
          class="pt-6"
        >
          <ul
            class="flex gap-3 flex-col"
          >
            <li>
              <a
                class="text-white-100 font-thin text-xs"
                href=""
              >
                Product
              </a>
            </li>
            <li>
              <a
                class="text-white-100 font-thin text-xs"
                href=""
              >
                Categories
              </a>
            </li>
            <li>
              <a
                class="text-white-100 font-thin text-xs"
                href=""
              >
                Shop
              </a>
            </li>
            <li>
              <a
                class="text-white-100 font-thin text-xs"
                href=""
              >
                Login
              </a>
            </li>
          </ul>
        </nav>
      </div>
      <div>
        <p
          class="text-sm font-bold text-white-100"
        >
          Shop
        </p>
        <nav
          class="pt-6"
        >
          <ul
            class="flex gap-3 flex-col"
          >
            <li>
              <a
                class="text-white-100 font-thin text-xs"
                href=""
              >
                T-shirt
              </a>
            </li>
            <li>
              <a
                class="text-white-100 font-thin text-xs"
                href=""
              >
                Jacket
              </a>
            </li>
            <li>
              <a
                class="text-white-100 font-thin text-xs"
                href=""
              >
                Shirt
              </a>
            </li>
            <li>
              <a
                class="text-white-100 font-thin text-xs"
                href=""
              >
                Jeans
              </a>
            </li>
          </ul>
        </nav>
      </div>
      <div>
        <p
          class="text-sm font-bold text-white-100"
        >
          Categories
        </p>
        <nav
          class="pt-6"
        >
          <ul
            class="flex gap-3 flex-col"
          >
            <li>
              <a
                class="text-white-100 font-thin text-xs"
                href=""
              >
                Men
              </a>
            </li>
            <li>
              <a
                class="text-white-100 font-thin text-xs"
                href=""
              >
                Children
              </a>
            </li>
            <li>
              <a
                class="text-white-100 font-thin text-xs"
                href=""
              >
                Women
              </a>
            </li>
          </ul>
        </nav>
      </div>
      <div>
        <p
          class="text-sm font-bold text-white-100"
        >
          Contact
        </p>
        <div
          class="flex gap-3 flex-col"
        >
          <address
            class="pt-6 not-italic"
          >
            <p
              class="text-xs font-undefined text-white-100"
            >
              mangcoding@gmail.com
            </p>
          </address>
          <div>
            <nav
              class="list-none flex gap-3"
            >
              <li>
                <a
                  href=""
                >
                  <svg
                    fill="none"
                    height="32"
                    viewBox="0 0 32 32"
                    width="32"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      height="31"
                      stroke="white"
                      width="31"
                      x="0.5"
                      y="0.5"
                    />
                    <g
                      clip-path="url(#clip0_2749_251)"
                    >
                      <path
                        d="M13.7869 23V16.4308H12V14.0656H13.7869V12.0454C13.7869 10.4578 14.8425 9 17.2747 9C18.2595 9 18.9877 9.09177 18.9877 9.09177L18.9303 11.3005C18.9303 11.3005 18.1876 11.2935 17.3772 11.2935C16.5002 11.2935 16.3596 11.6864 16.3596 12.3385V14.0656H19L18.8851 16.4308H16.3596V23H13.7869Z"
                        fill="white"
                      />
                    </g>
                    <defs>
                      <clippath
                        id="clip0_2749_251"
                      >
                        <rect
                          fill="white"
                          height="14"
                          transform="translate(12 9)"
                          width="7"
                        />
                      </clippath>
                    </defs>
                  </svg>
                </a>
              </li>
              <li>
                <a
                  href=""
                >
                  <svg
                    fill="none"
                    height="32"
                    viewBox="0 0 32 32"
                    width="32"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      height="31"
                      stroke="white"
                      width="31"
                      x="0.5"
                      y="0.5"
                    />
                    <g
                      clip-path="url(#clip0_2749_246)"
                    >
                      <path
                        d="M24.7793 10.7049C24.1123 10.9894 23.4049 11.1761 22.6806 11.2589C23.4351 10.8235 24.0144 10.134 24.2872 9.31247C23.5699 9.72228 22.7852 10.011 21.9669 10.1662C21.3003 9.48248 20.3507 9.05518 19.2997 9.05518C17.2817 9.05518 15.6455 10.6304 15.6455 12.5733C15.6455 12.849 15.6779 13.1175 15.7402 13.3751C12.7033 13.2283 10.0109 11.8277 8.2086 9.69918C7.89413 10.2188 7.71394 10.8232 7.71394 11.4679C7.71394 12.6885 8.35908 13.7653 9.3395 14.3962C8.75923 14.3787 8.19173 14.2278 7.68437 13.9561C7.68416 13.9708 7.68416 13.9856 7.68416 14.0004C7.68416 15.705 8.9437 17.1269 10.6152 17.4501C10.0772 17.591 9.51275 17.6117 8.96513 17.5104C9.43008 18.9082 10.7796 19.9253 12.3785 19.9537C11.1279 20.8973 9.55232 21.4598 7.84042 21.4598C7.54544 21.4598 7.25462 21.4431 6.96875 21.4106C8.58582 22.4088 10.5065 22.9913 12.57 22.9913C19.2912 22.9913 22.9665 17.6303 22.9665 12.9813C22.9665 12.8287 22.963 12.677 22.9559 12.5261C23.6713 12.0282 24.2887 11.4115 24.7793 10.7049Z"
                        fill="white"
                      />
                    </g>
                    <defs>
                      <clippath
                        id="clip0_2749_246"
                      >
                        <rect
                          fill="white"
                          height="14"
                          transform="translate(7 9)"
                          width="18"
                        />
                      </clippath>
                    </defs>
                  </svg>
                </a>
              </li>
              <li>
                <a
                  href=""
                >
                  <svg
                    fill="none"
                    height="32"
                    viewBox="0 0 32 32"
                    width="32"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      height="31"
                      stroke="white"
                      width="31"
                      x="0.5"
                      y="0.5"
                    />
                    <g
                      clip-path="url(#clip0_2749_241)"
                    >
                      <path
                        d="M11.5525 23V12.8795H8.15826V23H11.5525ZM9.8558 11.4969C11.0394 11.4969 11.7762 10.7198 11.7762 9.74862C11.7541 8.75556 11.0395 8 9.87826 8C8.71724 8 7.95801 8.75558 7.95801 9.74862C7.95801 10.7198 8.69457 11.4969 9.83364 11.4969H9.85569H9.8558ZM13.4311 23H16.8253V17.3482C16.8253 17.0458 16.8474 16.7436 16.937 16.5274C17.1824 15.923 17.7409 15.2971 18.6785 15.2971C19.9067 15.2971 20.3981 16.2252 20.3981 17.5857V22.9999H23.792V17.1969C23.792 14.0883 22.1175 12.6419 19.8843 12.6419C18.0533 12.6419 17.2494 13.6563 16.8027 14.3471H16.8254V12.8793H13.4312C13.4758 13.829 13.4312 22.9998 13.4312 22.9998L13.4311 23Z"
                        fill="white"
                      />
                    </g>
                    <defs>
                      <clippath
                        id="clip0_2749_241"
                      >
                        <rect
                          fill="white"
                          height="15"
                          transform="translate(8 8)"
                          width="16"
                        />
                      </clippath>
                    </defs>
                  </svg>
                </a>
              </li>
              <li>
                <a
                  href=""
                >
                  <svg
                    fill="none"
                    height="32"
                    viewBox="0 0 32 32"
                    width="32"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      height="31"
                      stroke="white"
                      width="31"
                      x="0.5"
                      y="0.5"
                    />
                    <g
                      clip-path="url(#clip0_2749_236)"
                    >
                      <path
                        d="M15.877 11.9571C13.6295 11.9571 11.8167 13.7606 11.8167 15.9965C11.8167 18.2325 13.6295 20.036 15.877 20.036C18.1246 20.036 19.9374 18.2325 19.9374 15.9965C19.9374 13.7606 18.1246 11.9571 15.877 11.9571ZM15.877 18.6227C14.4246 18.6227 13.2373 17.445 13.2373 15.9965C13.2373 14.5481 14.4211 13.3704 15.877 13.3704C17.333 13.3704 18.5168 14.5481 18.5168 15.9965C18.5168 17.445 17.3295 18.6227 15.877 18.6227ZM21.0506 11.7918C21.0506 12.3157 20.6265 12.734 20.1035 12.734C19.577 12.734 19.1564 12.3122 19.1564 11.7918C19.1564 11.2715 19.5805 10.8497 20.1035 10.8497C20.6265 10.8497 21.0506 11.2715 21.0506 11.7918ZM23.7398 12.7481C23.6797 11.486 23.39 10.368 22.4606 9.44692C21.5347 8.52583 20.411 8.23755 19.1423 8.17427C17.8348 8.10044 13.9158 8.10044 12.6083 8.17427C11.3431 8.23403 10.2194 8.52231 9.28998 9.44341C8.36059 10.3645 8.07434 11.4825 8.01074 12.7446C7.93653 14.0454 7.93653 17.9442 8.01074 19.245C8.07081 20.5071 8.36059 21.625 9.28998 22.5461C10.2194 23.4672 11.3396 23.7555 12.6083 23.8188C13.9158 23.8926 17.8348 23.8926 19.1423 23.8188C20.411 23.759 21.5347 23.4707 22.4606 22.5461C23.3864 21.625 23.6762 20.5071 23.7398 19.245C23.814 17.9442 23.814 14.0489 23.7398 12.7481ZM22.0507 20.6407C21.775 21.3297 21.2414 21.8606 20.5452 22.1383C19.5028 22.5497 17.0291 22.4547 15.877 22.4547C14.725 22.4547 12.2478 22.5461 11.2089 22.1383C10.5162 21.8641 9.98261 21.3332 9.70344 20.6407C9.28998 19.6036 9.3854 17.1426 9.3854 15.9965C9.3854 14.8504 9.29352 12.386 9.70344 11.3524C9.97908 10.6633 10.5127 10.1325 11.2089 9.85474C12.2513 9.44341 14.725 9.53833 15.877 9.53833C17.0291 9.53833 19.5063 9.44692 20.5452 9.85474C21.2379 10.129 21.7715 10.6598 22.0507 11.3524C22.4641 12.3895 22.3687 14.8504 22.3687 15.9965C22.3687 17.1426 22.4641 19.6071 22.0507 20.6407Z"
                        fill="white"
                      />
                    </g>
                    <defs>
                      <clippath
                        id="clip0_2749_236"
                      >
                        <rect
                          fill="white"
                          height="18"
                          transform="translate(8 7)"
                          width="16"
                        />
                      </clippath>
                    </defs>
                  </svg>
                </a>
              </li>
            </nav>
          </div>
        </div>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
